app.js

```diff
@@ -0,0 +1,84 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+
+var indexStatic;
+var router;
+var app = express();
+
+var startProdApp = function() {
+  console.log([
+    'Production mode enabled. Note that "npm install" is assumed to',
+    'have recently been run with NODE_ENV="production". If this is not',
+    'the case, some or all static assets may be out of date.'
+  ].join('\n'));
+  indexStaticWatcher.build(function(err, newIndexStatic) {
+    if (err) {
+      throw err;
+    }
+
+    console.log('Built server-side bundle.');
```

Comment:
> heh, sorry I think I phrased my question wrong - I understand the "dynamic" server being "dynamic" part :stuck_out_tongue: but I was wondering how template files/JS gets compiled from `.jsx`.  I assume React handles the compilation?

Ohhhh, got it! Yeah, that's actually powered by the infrastructure we added in #839, and it's different for development vs. production mode.

For development, we use the `index-static-watcher` thing to watch our JSX files (or any of the JS files they depend on) for changes and re-compile the server-side/html-generating JS bundle 
when needed.

For production, that whole thing is done only once (since we know our JSX/JS won't be tinkered with while the server is running) during the `postinstall` step.

Either way, the JSX/JS is ultimately bundled, via webpack, into a file in the `build` directory which the server loads (and potentially *re*loads, if we're in development mode). And that bundle is where `indexStatic.generate()` lives.